src/pytest_bdd/scenario.py

```diff
@@ -153,28 +153,32 @@ def get_fixture_path(fixture_def: FixtureDef) -> list[str]:
 
 def get_step_function(request: FixtureRequest, step: Step) -> StepFunctionContext | None:
     """Get the step function (context) for the given step.
-
     We first figure out what's the step fixture name that we have to inject.
-
     Then we let `patch_argumented_step_functions` find out what step definition fixtures can parse the current step,
     and it will inject them for the step fixture name.
-
-    Finally we let request.getfixturevalue(...) fetch the step definition fixture.
+    Finally, we let request.getfixturevalue(...) fetch the step definition fixture.
+    Data tables are considered if the step has them defined.
     """
     __tracebackhide__ = True
     bdd_name = get_step_fixture_name(step=step)
 
     with inject_fixturedefs_for_step(step=step, fixturemanager=request._fixturemanager, node=request.node):
         try:
-            return cast(StepFunctionContext, request.getfixturevalue(bdd_name))
+            context = cast(StepFunctionContext, request.getfixturevalue(bdd_name))
+
+            # Attach data_table if present
+            if step.data_table:
+                context.data_table = step.data_table
+
+            return context
         except pytest.FixtureLookupError:
             return None
 
 
 def _execute_step_function(
     request: FixtureRequest, scenario: Scenario, step: Step, context: StepFunctionContext
 ) -> None:
-    """Execute step function."""
+    """Execute step function, with support for data_table."""
```

Comment:
no need for this change, otherwise we would have to list all the things we support